### ISSUE TITLE
Push not registering on android, works on iOS

### ISSUE THREAD

**Issue by roel-sluper** [phonegap/phonegap-plugin-push#342]
Hello,

I've been working on my iOS app for a while now and had no trouble getting the push working with this plugin. I started development on its Android counterpart today but my device wont seem to register nor does is fail with any errors.

Im using the following code:

```javascript
App = {
    init: function() {
        alert('device ready');
        App.initPushNotification();
    },
    initPushNotification: function()
    {
        push = PushNotification.init({
            "android": {
                "senderID": "1136XXXXXXXX"
            },
            "ios": {
                "alert": "true",
                "badge": "true",
                "sound": "true",
                "clearBadge": "true"
            }
        } );
        push.on('registration', this.pushRegistration);
        push.on('notification', this.pushReceivedNotification);
        push.on('error', this.pushError);
    },
    pushRegistration: function(data) {
        console.log(data.registrationId);
        var deviceTokenFld = $('deviceToken');
        if (deviceTokenFld) {
            deviceTokenFld.set('value', data.registrationId);
            $('login-button').removeClass('disabled');
        }
    },
    pushReceivedNotification: function(data) {
        
        console.log(data);
        if (data.additionalData.deeplink) {
            var dialog = $('eventModal');
            $('eventModalBody').set('text', data.message);
            $('eventModalLink').set('href', data.additionalData.deeplink);
            dialog.show();
        }

    },
    pushError: function(e) {
        // e.message
        alert(e.message);
    }
};

document.addEventListener("deviceready", App.init, false);
```

I'm using Mootools as a JS framework. This code works fine on iOS but none of the callbacks are being fired on Android.

Any clue as to why this is happening? Or rather not happening ^^

**Comment by fredgalvao** [comment]
Is it a real device or is it an emulator? In the case it's an emulator, which target did you pick when creating the emulator image?

Also, can you grab the logcat output for your application? There should have a few hints as to what's happening in there. The javascript code seems just fine to me, so I'm guessing it's on native code or on the certificates somehow.

**Comment by roel-sluper** [comment]
I was testing on the emulator, Target Google API's, level 22.

This is the logcat output:
```logcat
11-17 09:41:05.266 2571-2571/? I/art: Not late-enabling -Xcheck:jni (already on)
11-17 09:41:06.793 2571-2597/nl.omines.sorama I/GMPM: App measurement is starting up
11-17 09:41:06.817 2571-2597/nl.omines.sorama E/GMPM: getGoogleAppId failed with status: 10
11-17 09:41:06.831 2571-2597/nl.omines.sorama E/GMPM: Uploading is not possible. App measurement disabled
11-17 09:41:07.223 2571-2571/nl.omines.sorama I/WebViewFactory: Loading com.android.webview version 39 (1737576-arm) (code 300001)
11-17 09:41:07.279 2571-2578/nl.omines.sorama W/art: Suspending all threads took: 9.089ms
11-17 09:41:07.417 2571-2571/nl.omines.sorama I/LibraryLoader: Time to load native libraries: 78 ms (timestamps 9769-9847)
11-17 09:41:07.450 2571-2571/nl.omines.sorama I/LibraryLoader: Expected native library version number "",actual native library version number ""
11-17 09:41:07.469 2571-2583/nl.omines.sorama I/art: Background sticky concurrent mark sweep GC freed 4820(290KB) AllocSpace objects, 4(64KB) LOS objects, 32% free, 750KB/1117KB, paused 1.439ms total 179.881ms
11-17 09:41:07.663 2571-2571/nl.omines.sorama V/WebViewChromiumFactoryProvider: Binding Chromium to main looper Looper (main, tid 1) {1f21a9c4}
11-17 09:41:07.665 2571-2571/nl.omines.sorama I/LibraryLoader: Expected native library version number "",actual native library version number ""
11-17 09:41:07.684 2571-2571/nl.omines.sorama I/chromium: [INFO:library_loader_hooks.cc(104)] Chromium logging enabled: level = 0, default verbosity = 0
11-17 09:41:07.731 2571-2571/nl.omines.sorama I/BrowserStartupController: Initializing chromium process, singleProcess=true
11-17 09:41:07.736 2571-2571/nl.omines.sorama W/art: Attempt to remove local handle scope entry from IRT, ignoring
11-17 09:41:07.854 2571-2614/nl.omines.sorama W/AudioManagerAndroid: Requires BLUETOOTH permission
11-17 09:41:07.882 2571-2571/nl.omines.sorama W/chromium: [WARNING:resource_bundle.cc(304)] locale_file_path.empty()
11-17 09:41:07.887 2571-2571/nl.omines.sorama I/chromium: [INFO:aw_browser_main_parts.cc(65)] Load from apk succesful, fd=31 off=46184 len=3037
11-17 09:41:07.894 2571-2571/nl.omines.sorama I/chromium: [INFO:aw_browser_main_parts.cc(78)] Loading webviewchromium.pak from, fd:32 off:229484 len:1089587
11-17 09:41:07.962 2571-2571/nl.omines.sorama E/chromium: [ERROR:gl_surface_egl.cc(165)] No suitable EGL configs found.
11-17 09:41:07.962 2571-2571/nl.omines.sorama E/chromium: [ERROR:gl_surface_android.cc(23)] GLSurfaceEGL::InitializeOneOff failed.
11-17 09:41:07.963 2571-2571/nl.omines.sorama E/chromium: [ERROR:browser_main_loop.cc(988)] GLSurface::InitializeOneOff failed
11-17 09:41:08.256 2571-2571/nl.omines.sorama W/chromium: [WARNING:data_reduction_proxy_settings.cc(331)] SPDY proxy OFF at startup
11-17 09:41:08.303 2571-2578/nl.omines.sorama W/art: Suspending all threads took: 37.549ms
11-17 09:41:08.370 2571-2583/nl.omines.sorama W/art: Suspending all threads took: 38.697ms
11-17 09:41:08.433 2571-2583/nl.omines.sorama I/art: Background partial concurrent mark sweep GC freed 981(94KB) AllocSpace objects, 2(32KB) LOS objects, 50% free, 1004KB/2028KB, paused 41.176ms total 361.313ms
11-17 09:41:08.492 2571-2571/nl.omines.sorama W/art: Attempt to remove local handle scope entry from IRT, ignoring
11-17 09:41:08.536 2571-2571/nl.omines.sorama W/AwContents: onDetachedFromWindow called when already detached. Ignoring
11-17 09:41:08.658 2571-2571/nl.omines.sorama D/SystemWebViewEngine: CordovaWebView is running on device made by: unknown
11-17 09:41:08.690 2571-2571/nl.omines.sorama W/art: Attempt to remove local handle scope entry from IRT, ignoring
11-17 09:41:08.690 2571-2571/nl.omines.sorama W/art: Attempt to remove local handle scope entry from IRT, ignoring
11-17 09:41:09.178 2571-2571/nl.omines.sorama D/gralloc_goldfish: Emulator without GPU emulation detected.
11-17 09:41:24.939 2571-2571/nl.omines.sorama I/Choreographer: Skipped 39 frames!  The application may be doing too much work on its main thread.
11-17 09:41:26.124 2571-2571/nl.omines.sorama I/Choreographer: Skipped 59 frames!  The application may be doing too much work on its main thread.
11-17 09:41:34.348 2571-2578/nl.omines.sorama W/art: Suspending all threads took: 40.349ms
```

**Comment by roel-sluper** [comment]
I've also got my hands on an physical device, same problem. Here is the logcat output of that:

```
11-17 12:47:31.647 1183-1183/? I/art: Late-enabling -Xcheck:jni
11-17 12:47:31.960 1183-1254/nl.omines.sorama I/GMPM: App measurement is starting up
11-17 12:47:31.966 1183-1254/nl.omines.sorama E/GMPM: getGoogleAppId failed with status: 10
11-17 12:47:31.968 1183-1254/nl.omines.sorama E/GMPM: Uploading is not possible. App measurement disabled
11-17 12:47:32.024 1183-1183/nl.omines.sorama I/WebViewFactory: Loading com.google.android.webview version 43.0.2357.121 (code 2357121)
11-17 12:47:32.073 1183-1183/nl.omines.sorama I/LibraryLoader: Time to load native libraries: 4 ms (timestamps 504-508)
11-17 12:47:32.074 1183-1183/nl.omines.sorama I/LibraryLoader: Expected native library version number "",actual native library version number ""
11-17 12:47:32.133 1183-1183/nl.omines.sorama V/WebViewChromiumFactoryProvider: Binding Chromium to main looper Looper (main, tid 1) {3d04aa50}
11-17 12:47:32.133 1183-1183/nl.omines.sorama I/LibraryLoader: Expected native library version number "",actual native library version number ""
11-17 12:47:32.134 1183-1183/nl.omines.sorama I/chromium: [INFO:library_loader_hooks.cc(112)] Chromium logging enabled: level = 0, default verbosity = 0
11-17 12:47:32.147 1183-1183/nl.omines.sorama I/BrowserStartupController: Initializing chromium process, singleProcess=true
11-17 12:47:32.148 1183-1183/nl.omines.sorama W/art: Attempt to remove local handle scope entry from IRT, ignoring
11-17 12:47:32.149 1183-1183/nl.omines.sorama E/SysUtils: ApplicationContext is null in ApplicationStatus
11-17 12:47:32.169 1183-1183/nl.omines.sorama W/chromium: [WARNING:resource_bundle.cc(286)] locale_file_path.empty()
11-17 12:47:32.169 1183-1183/nl.omines.sorama I/chromium: [INFO:aw_browser_main_parts.cc(63)] Load from apk succesful, fd=38 off=6843176 len=3647
11-17 12:47:32.170 1183-1183/nl.omines.sorama I/chromium: [INFO:aw_browser_main_parts.cc(76)] Loading webviewchromium.pak from, fd:39 off:7638088 len:1165478
11-17 12:47:32.176 1183-1183/nl.omines.sorama I/Adreno-EGL: <qeglDrvAPI_eglInitialize:410>: EGL 1.4 QUALCOMM build: AU_LINUX_ANDROID_LA.AF.1.1_RB1.05.00.02.006.020_msm8960_LA.AF.1.1_RB1__release_AU ()
11-17 12:47:32.176 1183-1183/nl.omines.sorama I/Adreno-EGL: OpenGL ES Shader Compiler Version: E031.25.03.06
11-17 12:47:32.176 1183-1183/nl.omines.sorama I/Adreno-EGL: Build Date: 05/22/15 Fri
11-17 12:47:32.176 1183-1183/nl.omines.sorama I/Adreno-EGL: Local Branch: mybranch10237523
11-17 12:47:32.176 1183-1183/nl.omines.sorama I/Adreno-EGL: Remote Branch: quic/LA.AF.1.1_rb1.18
11-17 12:47:32.176 1183-1183/nl.omines.sorama I/Adreno-EGL: Local Patches: NONE
11-17 12:47:32.176 1183-1183/nl.omines.sorama I/Adreno-EGL: Reconstruct Branch: AU_LINUX_ANDROID_LA.AF.1.1_RB1.05.00.02.006.020 + 9b2699f + 2215637 + 60aa592 + f2362e6 + 5c64f59 + 82411a1 + 89a0db6 + 6151be1 +  NOTHING
11-17 12:47:32.249 1183-1296/nl.omines.sorama W/AudioManagerAndroid: Requires BLUETOOTH permission
11-17 12:47:32.546 1183-1294/nl.omines.sorama W/chromium: [WARNING:data_reduction_proxy_config.cc(318)] SPDY proxy OFF at startup
11-17 12:47:32.581 1183-1183/nl.omines.sorama W/art: Attempt to remove local handle scope entry from IRT, ignoring
11-17 12:47:32.599 1183-1183/nl.omines.sorama W/AwContents: onDetachedFromWindow called when already detached. Ignoring
11-17 12:47:32.624 1183-1183/nl.omines.sorama D/SystemWebViewEngine: CordovaWebView is running on device made by: Sony
11-17 12:47:32.633 1183-1183/nl.omines.sorama W/art: Attempt to remove local handle scope entry from IRT, ignoring
11-17 12:47:32.633 1183-1183/nl.omines.sorama W/art: Attempt to remove local handle scope entry from IRT, ignoring
11-17 12:47:32.669 1183-1347/nl.omines.sorama D/OpenGLRenderer: Use EGL_SWAP_BEHAVIOR_PRESERVED: true
11-17 12:47:32.686 1183-1183/nl.omines.sorama D/Atlas: Validating map...
11-17 12:47:32.728 1183-1347/nl.omines.sorama I/OpenGLRenderer: Initialized EGL, version 1.4
11-17 12:47:32.734 1183-1347/nl.omines.sorama D/OpenGLRenderer: Enabling debug mode 0
11-17 12:47:32.928 1183-1183/nl.omines.sorama I/Timeline: Timeline: Activity_idle id: android.os.BinderProxy@3e83b55f time:3991363
11-17 12:47:33.156 1183-1183/nl.omines.sorama W/BindingManager: Cannot call determinedVisibility() - never saw a connection for the pid: 1183
11-17 12:47:33.807 1183-1183/nl.omines.sorama I/chromium: [INFO:CONSOLE(0)] "The specified value 't' is not a valid email address.", source: http://app.sorama.dev/login (0)
```

**Comment by macdonst** [comment]
@roel-sluper there is no indication that the PushPlugin is even called in your logcat. Please do `adb logcat | grep PushPlugin` as that will only output the logs from the plugin. Then start your app and try to register.

**Comment by roel-sluper** [comment]
When I run the app with the command open on the terminal it doesn't output anything.

```
MacBook-Pro-van-Roel:platform-tools roelsluper$ ./adb logcat | grep PushPlugin

```
So if I understand it correctly the plugin is never loaded?

**Comment by macdonst** [comment]
@roel-sluper it sure doesn't seem like it. I'd open `chrome://inspect/#devices` and see if `PushPlugin` is defined in the console.

**Comment by roel-sluper** [comment]
I don't have a PushPlugin object, I do however have a PushNotification object:

```
PushNotification
Object {}PushNotification: (options) {arguments: nullcaller: nulllength: 1name: ""prototype: PushNotification__proto__: Empty() {}<function scope>init: (options) {__proto__: Object
PushPlugin
VM105:2 Uncaught ReferenceError: PushPlugin is not defined
    at <anonymous>:2:1
    at Object.InjectedScript._evaluateOn (<anonymous>:895:140)
    at Object.InjectedScript._evaluateAndWrap (<anonymous>:828:34)
    at Object.InjectedScript.evaluate (<anonymous>:694:21)
```

**Comment by roel-sluper** [comment]
I did just find something odd. If i check ```cordova``` in the inspector I see something odd:

```json
{version: "3.9.2", platformVersion: "3.9.2", platformId: "ios", callbackId: 781854820, callbacks: Object…}
```

Why would my Android device have a platformId of ios?

**Comment by fredgalvao** [comment]
That is really, really strange. Can you check if the `JSON.stringify(window.device, null, 4)` object seems fine?

Also, can you please detail on your cordova environment? (`cordova -v`, `cordova platform ls`, `cordova plugin ls`)

**Comment by macdonst** [comment]
@roel-sluper check the cordova.js file in your android app it seems you got the ios one in there by mistake. If you look at the comment at the top of the file it'll tell you what version of the JS file you have in there.

**Comment by Menusito** [comment]
hello, im using android 4.4.2 with cross-walk, cli 5.2.0 and the event registration never fires.

**Comment by roel-sluper** [comment]
Sorry for the late response. I had a few days without out internet due to moving.

I looked into @macdonst suggestion, somehow the files must have gotten mixed up. I copied a new version of the android cordova.js and when I check the console on the remote inspector, the platformId now says android. 

When i fire the app up in the emulator I get an alert stating ```Empty registration ID received from GCM```, which I assume is due to it running on the emulator and not a physical device just as push doesn't work on the iOS simulator. My test device is currently charging and will test it as soon as it boots up. I'll get back to you when this is works fine. Thanks for the patience and help so far.
